ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -453,10 +432,63 @@ def test_ipa_healthcheck_log_rotate_file_exist_issue35(self):
         tasks.install_packages(self.master, HEALTHCHECK_PKG)
         assert self.master.transport.file_exists(HEALTHCHECK_LOG_ROTATE_CONF)
         cmd = self.master.run_command(
-            ['logrotate', '--debug', HEALTHCHECK_LOG_ROTATE_CONF]
+            ["logrotate", "--debug", HEALTHCHECK_LOG_ROTATE_CONF]
         )
         assert msg not in cmd.stdout_text
 
+    def test_ipa_dns_systemrecords_check(self):
+        """
+        This test ensures that the ipahealthcheck.ipa.idns check
+        displays the correct result when master and replica is setup
+        with integrated DNS.
+        """
+        SRV_MASTER_RECORDS = [
+            "_ldap._tcp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            "_kerberos._tcp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            "_kerberos._udp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            "_kerberos-master._tcp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            "_kerberos-master._udp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            "_kpasswd._tcp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            "_kpasswd._udp." + self.master.domain.name + ".:" +
+            self.master.hostname + ".",
+            '"' + self.master.domain.name.upper() + '"',
+            self.master.ip,
+        ]
+        SRV_REPLICA_RECORDS = [
+            "_ldap._tcp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            "_kerberos._tcp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            "_kerberos._udp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            "_kerberos-replica._tcp." + self.replicas[0].domain.name +
+            ".:" + self.replicas[0].hostname + ".",
+            "_kerberos-replica._udp." + self.replicas[0].domain.name +
+            ".:" + self.replicas[0].hostname + ".",
+            "_kpasswd._tcp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            "_kpasswd._udp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            '"' + self.replicas[0].domain.name.upper() + '"',
```

Comment:
The full key is \"EXAMPLE.TEST\" so you need those escape strings in there.

Also please use domain.realm.upper instead of domain.name.upper. They just happen to match for your case.